cmd/check-host-config/wait.go

```diff
@@ -0,0 +1,111 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/osbuild/images/cmd/check-host-config/mockos"
+)
+
+// runningWait emulates 'systemctl is-system-running --wait'
+// It blocks until the system reaches "running" or fails on other states.
+// It is requried for older versions of systemd that don't support the option (EL8)
+func runningWait(ctx context.Context, logger mockos.Logger) error {
+	// Unset DBUS_VERBOSE so subprocesses don't inherit it (breaks systemd output)
+	os.Unsetenv("DBUS_VERBOSE")
+
+	ticker := time.NewTicker(500 * time.Millisecond)
+	defer ticker.Stop()
+	lastLogTime := time.Now()
+
+	// Set up signal handling for OS termination signals
+	sigChan := make(chan os.Signal, 1)
+	signal.Notify(sigChan, syscall.SIGTERM, syscall.SIGINT)
+	defer signal.Stop(sigChan)
```

Comment:
Would it be nicer to have this set up in `main()` instead and have an overarching context starting form there that we use everywhere (and create new contexts on) that gets cancelled on these signals?

Reply:
Good observation, in fact, this is a less readable way of doing the same thing. I can utilize `NotifyContext` since context is being passed both to the wait and the main checking routines.